modules/administration_manual/pages/configuration/files/encryption/enabling-master-key-encryption.adoc

```diff
@@ -0,0 +1,127 @@
+[[enabling-encryption-from-the-command-line]]
+= Master Key Based Encryption
+Matthew Setter <matthew@matthewsetter.com>
+:experimental:
+:keywords: master key based encryption, encryption
+:description: With Master Key based encryption, there is only one key (or key pair) and all files are encrypted using that key pair. 
+
+With Master Key based encryption, there is only one key (or key pair) and all files are encrypted using that key pair. 
+This is highly recommended for new instances to avoid restrictions in functionality of user key encryption.
+
+== Enabling Master Key Based Encryption
+
+To enable Master Key based encryption requires two steps:
+
+. xref:enable-the-encryption-app[Enable the encryption app]
+. xref:enable-and-configure-master-key-based-encryption[Enable and configure User-Key based Encryption]
+
+[TIP] 
+====
+We strongly encourage you to put your server in single user mode before setting up encryption.
+
+To do so, run the following command:
+
+....
+sudo -u www-data php occ maintenance:singleuser --on
+....
+====
+
+[[enable-the-encryption-app]]
+=== Enable the Encryption App
+
+include::enable-encryption.adoc[leveloffset=+1,tag=enable-encryption-app-via-command-line]
+
+[[enable-and-configure-master-key-based-encryption]]
+=== Enable and Configure Master Key Based Encryption
+
+To enable and configure Master Key based encryption via the command-line, involves several commands.
+
+. Enable encryption.
+. Enable the master key.
++
+NOTE: The master key mode has to be setup in a newly created instance.
+. Encrypt all data.
+
+The following example shows how to carry out these steps.
+
+[source,console]
+....
+sudo -u www-data php occ encryption:enable
+sudo -u www-data php occ encryption:select-encryption-type masterkey
+sudo -u www-data php occ encryption:encrypt-all
+....
+
+[NOTE]
+====
+This command is not typically required, as the master key is often enabled at install time.
+As a result, when enabling it, there should be no data to encrypt.
+But, in case it’s being enabled after install, and the installation does have files which are unencrypted, encrypt-all can be used to encrypt them.
+====
+
+== View Current Encryption Status
+
+Get the current encryption status and the loaded encryption module:
+
+[source,console]
+....
+sudo -u www-data php occ encryption:status
+....
+
+This is equivalent to checking Enable server-side encryption on your Admin page:
+
+[source,console]
+....
+sudo -u www-data php occ encryption:enable
+Encryption enabled
+
+Default module: OC_DEFAULT_MODULE
+....
+
```

Comment:
Because suggested above to enable single user mode, I would add here to disable it now.